apis/__init__.py

```diff
@@ -1,13 +1,32 @@
-from apis.fund_api_routes import api as fund_api
+"""Contains the function which initialises the API with a cusdom DAO.
+
+Returns:
+    RestX API: A RestX Api ready to be attached to a flask app.
+"""
+from apis.fund_discovery_api.fund_discovery_routes import api as discovery_api
+from apis.fund_discovery_api.fund_discovery_routes import init_discovery_routes
+from apis.fund_store_api.fund_api_routes import api as fund_api
+from apis.fund_store_api.fund_api_routes import init_store_routes
 from flask_restx import Api
 
-api = Api(
-    title="Fund Store API",
-    version="0.1",
-    description=(
-        "An api for requesting infomation about funds from the fund store"
-    ),
-    # All API metadatas
-)
 
-api.add_namespace(fund_api, path="/funds")
+def load_dao(dummy_dao):
+
+    api = Api(
+        title="Fund Store API",
+        version="0.2",
+        description=(
+            "An api for requesting infomation about funds from the fund store"
+        ),
+        # All API metadatas
+    )
+
+    init_discovery_routes(dummy_dao)
+
+    init_store_routes(dummy_dao)
+
+    api.add_namespace(fund_api, path="/funds")
+
+    api.add_namespace(discovery_api, path="/funds/search")
+
+    return api
```

Comment:
Nice and concise this. Though wondering about passing this dao through multiple functions? Could it be defined globally perhaps (by env variable perhaps?). Also, the dao itself seems to be a representation of a single data object (in this case a fund), should the global variable more be a data storage *engine* object like a database type object that can manage multiple types of data object perhaps?

Reply:
Currently the DAO is an abstract implementation of a database ORM - we could technically replace the definition of the class with any other class that has a get and get_all method and everything would still work. I agree that in the future when the underlaying database is more concrete then we should write a more powerful class for interacting with it.